Packs/PAN-OS/Integrations/Panorama/Panorama.py

```diff
@@ -7716,6 +7717,19 @@ def all(self, filter_string: Optional[str] = None) -> Iterator[Union[Firewall, P
         for device in all_devices.values():
             yield device
 
+    def get_single_device(self, filter_string: str) -> Union[Firewall, Panorama]:
+        """
+        Returns JUST ONE device, based on the filter string, and errors if the filter returns more.
+        Safeguard for functions that should only ever operate on a single device.
+        :param filter_string: The exact ID of the device to return from the topology.
+        """
+        all_devices = {**self.firewall_objects, **self.panorama_objects}
+        if filter_string not in all_devices:
+            raise DemistoException(f"filter_str {filter_string} is not the exact ID of a host in this topology; " +
+                                   f"use a more specific filter string.")
+
+        return all_devices.get(filter_string)
```

Comment:
always better to go on positive rather than negative 

```suggestion
        if device := all_devices.get(filter_string):
           return device
        raise DemistoException(f"filter_str {filter_string} is not the exact ID of a host in this topology; " +
                                   f"use a more specific filter string.")

```